### PR TITLE
Hive integration

### PR DESCRIPTION
**Pull request by ASChilds**
Re-based from newest Magpie repo and updated with Hive functionality.

### CODE REVIEW

File: submission-scripts/script-templates/Makefile [hunk 1]
```diff
@@ -403,6 +432,27 @@ define create-hadoop-and-pig
...
+		magpie-zookeeper \
+		magpie-run-job-header-substitution \
+		magpie-run-job-$(DIST) > $(MAGPIE_HADOOP_AND_HIVE)
+	$(call common-substitution, ${MAGPIE_HADOOP_AND_HIVE},${JAVA_DEFAULT})
+	$(call common-additions, ${MAGPIE_HADOOP_AND_HIVE}, hadoop hive)
+	sed -i -e "s/MAGPIE_JOB_TYPE=\"\(.*\)\"/MAGPIE_JOB_TYPE=\"hive\"/" $(MAGPIE_HADOOP_AND_HIVE)
+	sed -i -e "s/HADOOP_SETUP=.*/HADOOP_SETUP=yes/" $(MAGPIE_HADOOP_AND_HIVE)
+	sed -i -e "s/HADOOP_MODE=\"\(.*\)\"/HADOOP_MODE=\"launch\"/" $(MAGPIE_HADOOP_AND_HIVE)
+	sed -i -e "s/HIVE_SETUP=.*/HIVE_SETUP=yes/" $(MAGPIE_HADOOP_AND_HIVE)
+endef
```
Comment (chu11) [inline submission-scripts/script-templates/Makefile]: remember to set ZOOKEEPER_SETUP=yes

File: doc/README.hive [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+For additional functionality/testing purposes you may wish to download the
+hive-testbench software from GitHub (https://github.com/hortonworks/hive-testbench.git)
+and use git to patch the beeline versions of the scripts into the hive-testbench directory.
+
+There are some instances where the metastore and database paths
+can become 'disjointed' Ex: the Metastore is configured for hdfs://node1/xxxx
+however the paths within the database report hdfs://node2/xxxx.  Sourcing the
+magpie-env.sh file and running the magpie-hive-update-table-paths.sh script should
+resolve this situation.
```
Comment (chu11) [inline doc/README.hive]: perhaps add some text to inform the user that they should run ```tpch-build.sh``` before trying testbench.

**Comment by chu11** [comment]
I think it ran to completion correctly?
```
*******************************************************
* Submitting Hive Testbench queries
*******************************************************
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/g/g0/achu/bigdata/apache-hive-2.3.2-bin/lib/log4j-slf4j-impl-2.6.2.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/usr/workspace/wsa/achu/bigdata/hadoop-2.9.0/share/hadoop/common/lib/slf4j-log4j12-1.7.25.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.apache.logging.slf4j.Log4jLoggerFactory]
Connecting to jdbc:hive2://catalyst179:10000
Connected to: Apache Hive (version 2.3.2)
Driver: Hive JDBC (version 2.3.2)
Transaction isolation: TRANSACTION_REPEATABLE_READ
4 rows selected (0.155 seconds)
Beeline version 2.3.2 by Apache Hive
Closing: 0: jdbc:hive2://catalyst179:10000
Check shows: | tpch_flat_orc_2  |
Running /g/g0/achu/bigdata/apache-hive-2.3.2-bin/bin/beeline -u jdbc:hive2://catalyst179:10000/tpch_flat_orc_2 -i /g/g0/achu/bigdata/hive-testbench/sample-queries-tpch/testbench.settings -f /g/g0/achu/bigdata/hive-testbench/sample-queries-tpch/tpch_query1.sql, logging to /tmp/achu/hive/testhive/348042/log/348042_qry1_results.txt
Running /g/g0/achu/bigdata/apache-hive-2.3.2-bin/bin/beeline -u jdbc:hive2://catalyst179:10000/tpch_flat_orc_2 -i /g/g0/achu/bigdata/hive-testbench/sample-queries-tpch/testbench.settings -f /g/g0/achu/bigdata/hive-testbench/sample-queries-tpch/tpch_query2.sql, logging to /tmp/achu/hive/testhive/348042/log/348042_qry2_results.txt
Running /g/g0/achu/bigdata/apache-hive-2.3.2-bin/bin/beeline -u jdbc:hive2://catalyst179:10000/tpch_flat_orc_2 -i /g/g0/achu/bigdata/hive-testbench/sample-queries-tpch/testbench.settings -f /g/g0/achu/bigdata/hive-testbench/sample-queries-tpch/tpch_query3.sql, logging to /tmp/achu/hive/testhive/348042/log/348042_qry3_results.txt
Running /g/g0/achu/bigdata/apache-hive-2.3.2-bin/bin/beeline -u jdbc:hive2://catalyst179:10000/tpch_flat_orc_2 -i /g/g0/achu/bigdata/hive-testbench/sample-queries-tpch/testbench.settings -f /g/g0/achu/bigdata/hive-testbench/sample-queries-tpch/tpch_query4.sql, logging to /tmp/achu/hive/testhive/348042/log/348042_qry4_results.txt
*******************************************************
* Completed Hive Testbench
* Finish: Thu Sep 27 15:00:50 PDT 2018 
*******************************************************
```
```

**Comment by chu11** [comment]
assuming the above is correct output, I think we can merge.  Just two other things.

1) Could you add Hive version support to ```README.md``` and ```README```

2) double check for cornercase & functionality hive tests.  There are a few new areas that may have been missed when you forward ported the old patch series.

**Comment by ASChilds** [comment]
I've updated per your comments above, I think I've got all the testsuite cases taken care of. 
One other check for the testbench is to view results of the queries run (under /tmp/achu/hive/testhive/348042/log) and check for errors/results.  Otherwise it looks successful to me

**Pull request merged**